autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillSettingsViewModel.kt

```diff
@@ -458,3 +471,19 @@ class AutofillSettingsViewModel @Inject constructor(
         object ShowManualCredentialMode : CredentialModeCommand()
     }
 }
+
+interface WebUrlIdentifier {
+    fun isLikelyAUrl(domain: String?): Boolean
+}
+
+@ContributesBinding(ActivityScope::class)
+class RegexBasedUrlIdentifier @Inject constructor() : WebUrlIdentifier {
+
+    override fun isLikelyAUrl(domain: String?): Boolean {
+        if (domain.isNullOrBlank()) {
+            return false
+        }
+
+        return Patterns.WEB_URL.matcher(domain).matches()
+    }
+}
```

Comment:
NIT: You are injecting the interface in the ViewModel so maybe it's worth it to have it in a different file with the implementation? Having it in the ViewModel maybe it's quite confusing in case we want to reuse it? Up to you though 👍